### PR TITLE
Handle empty file paths in cucumber pull

### PR DESCRIPTION
**Pull request by PierrickTassery**
## Summary

This keeps the `pull` workflow from crashing when the API returns an empty file path in `data.files`.

When the file path is empty, the CLI now generates a fallback filename from the Gherkin `Feature:` title and writes the file there instead of trying to write the target directory itself.

Refs #70.

## Why

`check-cucumber pull` currently assumes every key in `data.files` is a writable filename. If the API returns an empty key:

```js
files: {
  "": "Feature: Missing Source Path\n..."
}
```

then `path.join(targetDir, "")` resolves to the target directory and `fs.writeFileSync` fails with:

```text
EISDIR: illegal operation on a directory, open '.'
```

## Changes

- Add a minimal fallback filename generator for empty file paths.
- Use the feature title as the fallback filename base.
- Add a regression test for an empty file path response.

This PR intentionally does not try to solve duplicate source-path collisions; that requires an API-side change because missing feature contents are not present in `data.files`.

## Testing

```bash
npm test
```

Result:

```text
97 passing
```

### CODE REVIEW

File: pull.js [hunk 1]
```diff
@@ -76,6 +78,31 @@ class Pull {
   }
 
 
+  /**
+   * Generate a fallback filename when the server returns an empty path.
+   * @param {string} content - Feature file content
+   * @param {Set<string>} usedFileNames - File names already reserved for this pull
+   * @returns {string} Safe fallback file name
+   */
+  fallbackFileName(content, usedFileNames) {
```
Comment (DavertMik) [inline pull.js:87]: @DenysKuchma please check if we can come up wiht better implementaton
regexes look fragile, we need to think about it

**Comment by DavertMik** [comment]
Thanks I will ask our developer to review and merge or provide alternative solution
